File: 1706-where-will-the-ball-fall/1706-where-will-the-ball-fall.cpp
class Solution {
public:
    vector<int> findBall(vector<vector<int>>& grid) {
        
        int ROWS = grid.size(), COLS = grid[0].size();
        
        vector<int> result(COLS, -1);
        
        for(int c=0; c<COLS; c++){
            
            int x=0, y=c;
            while(true){
                
                if(x >= ROWS){
                    
                    if(y>=0 && y<COLS) result[c] = y;
                    break;
                }
                else{
                    if(y+1<COLS && grid[x][y] == 1 && grid[x][y+1] == 1){
                        x++;
                        y++;
                    }
                    else if(y-1>=0 && grid[x][y] == -1 && grid[x][y-1] == -1){
                        x++;
                        y--;
                    }
                    else break;
                }
            }
        }
        
        return result;
    }
};

File: 1706-where-will-the-ball-fall/1706-where-will-the-ball-fall.py
class Solution:
    def findBall(self, grid: List[List[int]]) -> List[int]:
        
        ROWS, COLS = len(grid), len(grid[0])
        
        res = [-1]*COLS
        
        for i in range(COLS):
            x, y = 0, i
            while True:
                if x >= ROWS:
                    if y>=0 and y<COLS:
                        res[i] = y
                    break
                else:
                    if y+1<COLS and grid[x][y]==1 and grid[x][y+1]==1:
                        x += 1
                        y += 1
                    elif y-1>=0 and grid[x][y]==-1 and grid[x][y-1]==-1:
                        x += 1
                        y -= 1
                    else:
                        break
        return res
    